include/boost/process/detail/posix/handles.hpp

```diff
@@ -22,18 +22,22 @@ inline std::vector<native_handle_type> get_handles(std::error_code & ec)
 {
     std::vector<native_handle_type> res;
 
-    auto dir = ::opendir("/dev/fd");
-    auto my_fd = ::dirfd(dir);
+    std::unique_ptr<DIR, decltype(&::closedir)> dir{
```

Comment:
It's better to have a struct function object that calls `closedir`. It should avoid the warnings and is more optimal.

Reply:
Alright, will add that once all the tests pass. Thanks for reviewing!